src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -56,14 +56,23 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
 
             if ((object)boundRHS.Type == null)
```

Comment:
How do we distinguish between the case:

- RHS is `(null, null)` and hence has no type
- RHS is in error and hence has no type.  

In the latter case I assume we wouldn't want to go down this code path.